### PR TITLE
[SPARK-35775][SQL][TESTS] Check all year-month interval types in aggregate expressions

### PR DESCRIPTION
**Pull request by sarutak**
### What changes were proposed in this pull request?

This PR adds test to check `sum` and `avg` works with all the `YearMonthInterval` types.

### Why are the changes needed?

To ensure the results of aggregations are what is expected.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

New test.

### CODE REVIEW

**Review by MaxGekk — COMMENTED** [review]
LGTM except of a comment.

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala [hunk 1]
```diff
@@ -1115,34 +1116,47 @@ class DataFrameAggregateSuite extends QueryTest
...
-    checkAnswer(sumDF, Row(Period.of(2, 5, 0), Duration.ofDays(0)))
+    val sumDF = df.select(sum($"year-month"), sum($"year"), sum($"month"), sum($"day-time"))
+    checkAnswer(sumDF,
+      Row(Period.of(2, 5, 0), Period.ofYears(27), Period.of(1, 1, 0), Duration.ofDays(0)))
     assert(find(sumDF.queryExecution.executedPlan)(_.isInstanceOf[HashAggregateExec]).isDefined)
     assert(sumDF.schema == StructType(Seq(
-      // TODO(SPARK-35775): Check all year-month interval types in aggregate expressions
       StructField("sum(year-month)", YearMonthIntervalType()),
+      StructField("sum(year)", YearMonthIntervalType(0, 0)),
+      StructField("sum(month)", YearMonthIntervalType(1, 1)),
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala:1144]: change this too, please.

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala [hunk 2]
```diff
@@ -1115,34 +1116,47 @@ class DataFrameAggregateSuite extends QueryTest
...
+      Row(1, Period.ofMonths(10), Period.ofYears(8), Period.ofMonths(10), Duration.ofDays(10)) ::
+      Row(2, Period.ofMonths(1), Period.ofYears(1), Period.ofMonths(1), Duration.ofDays(1)) ::
+      Row(3, Period.of(1, 6, 0), Period.ofYears(18), Period.ofMonths(2), Duration.ofDays(-11)) ::
+      Nil)
     assert(find(sumDF2.queryExecution.executedPlan)(_.isInstanceOf[HashAggregateExec]).isDefined)
     assert(sumDF2.schema == StructType(Seq(StructField("class", IntegerType, false),
-      // TODO(SPARK-35775): Check all year-month interval types in aggregate expressions
       StructField("sum(year-month)", YearMonthIntervalType()),
+      StructField("sum(year)", YearMonthIntervalType(0, 0)),
+      StructField("sum(month)", YearMonthIntervalType(1, 1)),
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala:1159]: `YearMonthIntervalType(1, 1)` -> `YearMonthIntervalType(MONTH, MONTH)`

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala (hunk 1 above)
Comment (sarutak) [reply sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala]: Sorry, I've noticed after I pushed and already fixed.

**Comment by sarutak** [comment]
retest this please.

**Review by MaxGekk — APPROVED** [review]
GA passed. Merging to master.
Screenshot 2021-06-21 at 16 45 29
Thank you, @sarutak .

**Pull request closed without merging**